test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1997,30 +1982,18 @@ public String getMasterName(@Nullable String viaNode) {
     }
 
     /**
-     * Returns the name of the selected health node in the cluster. If there is no selected health node it returns null.
+     * @return the name of a random node in a cluster
      */
-    public String getHealthNodeName() {
-        return getHealthNodeName(null);
+    public String getRandomNodeName() {
+        return getNodeNameThat(ignored -> true);
     }
 
     /**
-     * Returns the name of the selected health node in the cluster and executes the request via the node specified
-     * in the viaNode parameter. If viaNode isn't specified a random node will be picked to send the request to. If
-     * there is no selected health node it returns null.
+     * @return the name of a random node in a cluster that match the {@code predicate}
      */
-    public String getHealthNodeName(@Nullable String viaNode) {
```

Comment:
Pinging @gmarouli about the removal of this unused code - not sure if the intention was to use it in a future PR or not, but if so then that PR will have to bring it back.